sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -369,6 +370,22 @@ datetimePart
     | complexDateTimePart
     ;
 
+nestedFunction
+    : NESTED LR_BRACKET
+    (nestedField | nestedField COMMA nestedPath)
+    RR_BRACKET
+    ;
+
+nestedField
+    : nestedIdent DOT nestedIdent (DOT nestedIdent)*
+    | stringLiteral
+    ;
+
+nestedPath
+    : nestedIdent (DOT nestedIdent)*
+    | stringLiteral
+    ;
```

Comment:
Can we just treat NESTED as a regular function and avoid these rules? Because I don't see special syntax. Mostly we add separate function rule because special syntax, ex. position function use IN keyword, relevancy function uses key-value attributes.

Reply:
To match the legacy implementation the NESTED function is displayed as the FIELD only in the column. So the function call `nested(message.info, message)` is displayed as `message.info`. If matching the legacy in this way is not important we could remove these rules.

Reply:
Yeah, I'm thinking whether we should follow it. Because in any other case, we don't do this in V2.

Reply:
I have updated to make the `nested` function generic. Now the column value is displayed as other functions in the V2 engine.